internal/gh/projects.go

```diff
@@ -0,0 +1,30 @@
+package gh
+
+// ProjectsV1Support provides type safety and readability around whether or not Projects v1 is supported
+// by the targeted host.
+//
+// It is a sealed type to ensure that consumers must use the exported ProjectsV1Supported and ProjectsV1Unsupported
+// variables to get an instance of the type.
+type ProjectsV1Support interface {
+	sealed()
+}
+
+type projectsV1Supported struct{}
+
+func (projectsV1Supported) sealed() {}
+
+type projectsV1Unsupported struct{}
+
+func (projectsV1Unsupported) sealed() {}
+
+var (
+	ProjectsV1Supported   ProjectsV1Support = projectsV1Supported{}
+	ProjectsV1Unsupported ProjectsV1Support = projectsV1Unsupported{}
+)
```

Comment:
Let me type out how I think this is working and you can correct me if I'm wrong or missed anything:

We are setting an Env var, `PROJECTS_V1_SUPPORTED`, with the feature detector. Then, the code parses that env var and, if that env var is set to `true`, will set our `ProjectV1Support` struct equal to the `ProjectsV1Supported` struct. 

Then, we use the struct to check equivalence in the relevant code, checking if the `ProjectsV1Support` value is equal to one struct or the other, returning the boolean we were relying on earlier for this decision.

What did I miss?

Reply:
Hmmm not sure this is quite right.

Ultimately, what we're passing around the application is a `ProjectsV1Support` interface type. This type is satisfied by two concrete types `projectsV1Supported` and projectsV1Unsupported`. In TypeScript you could imagine:

```typescript
type projectsV1Supported = {
   _tag: "projectsV1Supported"
}

type projectsV1Unsupported = {
   _tag: "projectsV1Unsupported"
}

type ProjectsV1Support = projectsV1Supported | projectsV1Unsupported

const projectsV1Supported: ProjectsV1Support = { _tag: "projectsV1Supported" }
const projectsV1Unsupported: ProjectsV1Support = { _tag: "projectsV1Unsupported" }
```

So everything can work with a `ProjectsV1Support` type that can be either side of the union. These can be compared by saying "is this thing I have equal to `projectsV1Supported`?" or "is this thing I have equal to `projectsV1Unsupported`?"

So how do we end up with one of these `ProjectsV1Support` types? Well, the feature detector introspects the GQL schema and if the `projects` field exists then we reference `projectsV1Supported` and if it doesn't we reference `projectsV1Unsupported`:

https://github.com/cli/cli/blob/534cbb755ab0b0918b3667e9e1d437d1877bcb62/internal/featuredetection/feature_detection.go#L229-L235

However, for the sake of developer testing, Andy decided to add an environment variable so that we can **pretend** the API supports or doesn't support projectsV1. This shortcircuits the detector so that it doesn't introspect the API. Instead, it uses the environment variable value, converts it to a boolean and then uses the `ParseProjectsV1Support` to convert that boolean to one of the concrete types that is satisfying `ProjectsV1Support`:

https://github.com/cli/cli/blob/534cbb755ab0b0918b3667e9e1d437d1877bcb62/internal/featuredetection/feature_detection.go#L208-L214

Hope this makes sense!

Reply:
Oh, yeah it totally does. I forgot about the short circuit and had assumed that the feature detector was setting that 😅 Thanks for the clarification